src/main.ts

```diff
@@ -24,17 +25,22 @@ async function run(): Promise<void> {
 
     if (!issueKey) {
       console.log(`JIRA key was not found`);
-      process.exit(0);
+      if (FAIL_PR_WHEN_JIRA_ISSUE_NOT_FOUND === 'true') {
+        console.log('Setting PR status as failed');
+        process.exit(1);
+      } else {
+        console.log('Ignore the errors and set PR status as success');
+        process.exit(0);
+      }
     }
 
     console.log(`JIRA key -> ${issueKey}`);
 
-    const details = await jiraConnector.getTicketDetails(issueKey);
-    await githubConnector.updatePrDetails(details);
+    await jiraConnector.getTicketDetails(issueKey, FAIL_PR_WHEN_JIRA_ISSUE_NOT_FOUND);
   } catch (error) {
+    console.log('Something went wrong!');
     console.log({ error });
     core.setFailed(error.message);
```

Comment:
core.setFailed is important in any action failure handling and should be used instead of process.exit(1)

Reply:
These are now set when the varible evaluates to true. If not, we will not fail the PR.